aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1604,6 +1606,125 @@ Tensor& linalg_norm_out(Tensor& result, const Tensor& self, std::string ord, opt
   return linalg_norm_out_impl(result, self, c10::nullopt, ord, opt_dim, keepdim, opt_dtype);
 }
 
+Tensor _linalg_cond_exception_helper(const Tensor& self) {
+  // For batched input if at least one matrix in the batch is not invertible,
+  // we can't get the result for all other (possibly) invertible matrices in the batch without an explicit for loop.
+  // This should change when at::inverse works with silent errors
+  if (self.dim() > 2) {
+    TORCH_CHECK(false,
+      "One or more matrices in the batch was not invertible! "
+      "linalg_cond does not support yet this case.");
```

Comment:
Let's remove this second line.

Reply:
Follow-up question, if the intent is to turn inf for every value, should this be a warning and not a check?

Maybe something like...

```
TORCH_WARN("One or more matrices in the batch was not invertible! This will cause the entire result to be filled with infs.");
```

?

Reply:
We've decided here https://github.com/pytorch/pytorch/pull/45832#discussion_r520748109 that since returning inf for every matrix is not correct we should raise an error. The batched case will be properly implemented in the future once it's possible to call `at::inverse` with silent errors (this should be possible after https://github.com/pytorch/pytorch/pull/48261 is ready, similarly to how it's now possible to call `at::solve_out_info` without error checks [here](https://github.com/pytorch/pytorch/pull/48456/files#diff-76df047bca5c2d7a540bdcc4023d68641c4f15638d83c5c874d833b03f9636b9R437-R439).

Reply:
OK; I'm conviced.